LazyProxy/LazyProxyBuilder.cs

```diff
@@ -111,6 +111,15 @@ private static Type DefineProxyType(Type type)
 
         private static TypeBuilder AddInterface(this TypeBuilder typeBuilder, Type type)
         {
+            if (type.IsGenericTypeDefinition)
+            {
+                var parameterNames = type.GetGenericArguments()
```

Comment:
We need to copy generic type constraints probably

Reply:
I'm on it